lib/kennel.rb

```diff
@@ -58,12 +61,19 @@ def generate
     end
 
     def plan
-      syncer.plan
+      the_plan = syncer.plan
+      PlanResults.new(
+        plan: the_plan
+      )
     end
 
     def update
```

Comment:
could `update` maybe get a plan ?
```suggestion
    def update(plan)
```

Reply:
hmmm I think this will need more refactor since there is lots of instance state ...

Reply:
Small changes, future PRs. Yes the instance state in kennel is pretty messy to work with. I have a vague plan for how to address it, and in no way does it fit in this PR.